src/gateway/sharding/shard_runner.rs

```diff
@@ -475,10 +455,10 @@ impl ShardRunner {
     async fn request_restart(&mut self) {
         debug!("[ShardRunner {:?}] Requesting restart", self.shard.shard_info());
 
-        self.update_manager().await;
+        self.update_runner_info().await;
```

Comment:
I don't see a reason why the runner info should be updated before sending a message that removes this info from the runners map, can you remove this?

Reply:
Done